sync-core/src/main/java/com/cloudant/sync/replication/BasicPullStrategy.java

```diff
@@ -377,12 +381,18 @@ private int processOneChangesBatch(ChangesResultWrapper changeFeeds)
         return changesProcessed;
     }
 
-    private String getReplicationId() {
-        if(filter == null) {
-            return this.sourceDb.getIdentifier() ;
-        } else {
-            return this.sourceDb.getIdentifier() + "?" + filter.toQueryString();
+    public String getReplicationId() {
+        HashMap<String, String> dict = new HashMap<String, String>();
+        dict.put("source", this.sourceDb.getIdentifier());
+        dict.put("target", this.targetDb.getDbCore().getPublicIdentifier());
+        if(filter != null) {
+            dict.put("filter", this.filter.toQueryString());
         }
+        // get raw SHA-1 of dictionary
+        byte[] sha1Bytes = Misc.getSha1(new ByteArrayInputStream(JSONUtils.serializeAsBytes(dict)));
+        // return SHA-1 as a hex string
+        byte[] sha1Hex = new Hex().encode(sha1Bytes);
+        return new String(sha1Hex);
```

Comment:
Oh, I see. When you use the local/remote rather than source/target as the identifiers, you need to include push to differentiate between the push and the pull case, whereas with source/target you get that automatically.